Grid table CSS refactor

stitot opened the pull request:
Enhanced CSS to allow theming of individual columns, cells etc. and added new hover opacity variables.

The following has been changed:
1. Complete rewrite of table elements variables and theming system
2. Introduced `--hcg-hover-opacity` for simpler and more unified hover state theming
3. Deprecated all `hovered` and `synced` variables and introduced new ones
4. Removed possibility to theme hover borders (will add docs on how to accomplish this moving forward)
5. Organized all `<table>` CSS elements in "base elements" and "themeable elements" as a head start on modularization. "themeable elements" can now be completely removed to allow for complete custom styling instead theming.
6. Added possibility to theme specific columns, rows, cells and headers using regular variables, with no need for `!important`
7. Added consistent theming of even/odd rows AND columns
8. Header is now considered part of the column, so all column styles now also apply to header unless overridden
9. Added a new theming basic sample (WIP)
10. Updated relevant demos to use new variables with no need for `!important`
11. Updated docs and added placeholder article for conditional theming (theming of individual rows, columns etc.)

Remains:
1. Testing 
2. Conditional theming article

Next steps:
This PR only concerns table elements (table, thead, th, td etc.). Child elements, such as icons, resizer etc. are not covered and will need a refactor as well, but no breaking changes are introduced in this PR (I hope).

sebastianbochan commented:
Mobile:
Headers disappears, when switching to mobile
![header](https://github.com/user-attachments/assets/ad21f1a1-b2fe-4d5a-b6e5-118f6f1d5d7d)

Otherwise looks good!

stitot commented:
@sebastianbochan :

1. Fixed
2. Fixed
3: IMO centered is harder to read, so leaving it as is
4. Fixed
5. This will be fixed in another PR
6. This internal sample has no dark mode for now. Will fix a little later, but shouldn't stop approval of this PR.
7. Fixed

stitot commented:
> Maybe it's just a CSS cache when changing the content of the same files

Could be, but I'm quite sure I actually observed that `no-width` didn't get removed from DOM when I increased size of window again. That is not CSS cache. Hm...

sebastianbochan commented:
All looks good, only what's left is the `no-width` issue.

ddragula reviewed (approved):
Looks good!
